lib/virt_autotest/esxi_utils.pm

```diff
@@ -25,46 +25,84 @@ our @EXPORT = qw(
   esxi_vm_public_ip
   get_host_timestamp
   disable_vm_time_synchronization
+  recover_vmx_config
 );
 
+my $hypervisor = get_var('HYPERVISOR') // 'esxi7.qa.suse.cz';
+
 sub esxi_vm_get_vmid {
     my $vm_name = shift;
-    my $vim_cmd = "vim-cmd vmsvc/getallvms | grep -i $vm_name | awk -F' ' '{print \$1}'";
-    my (undef, $vmid) = console('svirt')->run_cmd($vim_cmd, domain => 'sshVMwareServer', wantarray => 1);
-    chomp($vmid);
+    my $vim_cmd = "vim-cmd vmsvc/getallvms | grep -i $vm_name | cut -d' ' -f1";
+    my $vmid;
```

Comment:
Initialize ```my $vmid = 0 or ''``` ?

Reply:
It declares the variable $vmid, and the default value is 'undef', isn't it?

Reply:
@nanzhg  Hi Nan, you can have a look at https://progress.opensuse.org/issues/120750. We used to have dedicated ticket to fix uninitialized values to stop perl warnings. Recommend to follow this good habit.

Reply:
@alice-suse OK, there is only one warning I can see in the following job log, and I've fixed it.

https://openqa.suse.de/tests/10621486/logfile?filename=autoinst-log.txt
`Use of uninitialized value in numeric eq (==) at /var/lib/openqa/pool/3/os-autoinst-distri-opensuse/tests/virt_autotest/esxi_open_vm_tools.pm line 106.`

Warning statement:
`    } elsif ($powerops_ret == undef) {`
Fixed with:
`    } elsif (!defined $powerops_ret) {`